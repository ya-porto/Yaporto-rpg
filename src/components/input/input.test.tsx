import React from 'react';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import {mount, shallow, configure} from 'enzyme';
import {Input, IInputCompProps} from './index';

configure({adapter: new Adapter()});
describe('Input test', () => {
	const data: IInputCompProps = {
		value: 'invalid value',
		type: 'password',
		placeholder: 'Пароль',
		name: 'password',
		validation: {
			required: true,
			password: true
		},
		onChange: () => null
	};
	const template = <Input {...data} />;

	it('Snapshot', () => {
		const tree = renderer.create(template).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('Props', () => {
		const comp = mount(template);

		// @ts-ignore
		expect(comp.props().value).toMatch(data.value);
		expect(comp.props().type).toMatch(data.type);
		// @ts-ignore
		expect(comp.props().placeholder).toMatch(data.placeholder);
		// @ts-ignore
		expect(comp.props().name).toMatch(data.name);
		// @ts-ignore
		expect(comp.props().validation).toMatchObject(data.validation);
	});

	it('Validation', () => {
		const input = shallow(template);

		input.find('input').simulate('focus');
		// eslint-disable-next-line no-unused-expressions
		expect(input.state().error.isShow).toBeFalsy;

		input.find('input').simulate('blur');
		// eslint-disable-next-line no-unused-expressions
		expect(input.state().error.isShow).toBeTruthy;
	});
});
