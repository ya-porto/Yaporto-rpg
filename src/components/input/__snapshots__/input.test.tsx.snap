// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input test Snapshot 1`] = `
<div
  className="field "
>
  <input
    className="form__field"
    name="password"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Пароль"
    type="password"
    value="invalid value"
  />
  <label
    className="form__label"
    htmlFor="password"
  >
    Пароль
  </label>
</div>
`;
